tests/test_back.js

```diff
@@ -151,30 +151,26 @@ tap.test('nockBack dryrun tests', function (nw) {
   nockBack.fixtures = __dirname + '/fixtures';
   nockBack.setMode('dryrun');
 
-  nw.test('goes to internet even when no nockBacks are running', {skip: process.env.AIRPLANE}, function(t) {
-    var req = http.request({
-        host: "www.amazon.com"
-      , path: '/'
-      , port: 80
-      }, function(res) {
-
-        res.on('data', function() {
-          //node v 0.10 requires this listener
-        });
-        t.ok([200, 301, 302].indexOf(res.statusCode) >= 0);
-        t.end();
-
-      });
-
-    req.on('error', function(err) {
-
-      //  This should never happen.
-      t.assert(false);
-      t.end();
-
-    });
-
-    req.end();
+  nw.test('goes to internet even when no nockBacks are running', function(t) {
+    t.plan(2)
+
+    const server = http.createServer((request, response) => {
+      t.pass('server receives request')
+      response.writeHead(200)
+      response.end()
+    })
+    server.listen(0, () => {
```

Comment:
Will the tests run in parallel? Because, it would be tough to figure out the behaviour of our tests in such a case if all of them use port 0 to setup the local server.

Do you think it makes sense to persist the server across tests, if they all are in the same file?

I also have some concerns with the usage of port 0 which is reserved for TCP/IP networking to do some fancy socket programming things. We can probably just use good 'ol 80(80) if we decide to persist the server?

Reply:
setting port to `0` lets the server pick an available port, so there won’t be any conflicts, see https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback

Turns out the `0` argument can be omitted altogether, I’ll give this a try :)